docs/v1-proposed-api.md

```diff
@@ -0,0 +1,145 @@
+# Proposed Changes for Hive V1 API
+
+## ClusterDeployment
+
+ 1. Require an InstallConfig ConfigMap, referenced in a new ClusterDeployment.Spec field, to pass directly to installer.
+   * GOAL: Allow quicker exposure to new Installer features without needing to mirror in Hive's API.
+   * GOAL: Remaining ClusterDeployment.Spec should be things that can be modified post-install.
+   * Should Hive parse the InstallConfig to obtain information it needs, eliminating the need for the caller to duplicate information in ClusterDeployment and InstallConfig?
+     * Or should Hive be explicitly given all the information it needs to adopt and maintain a cluster in it's ClusterDeployment, requiring the caller to duplicate a fair portion of InstallConfig in each.
+     * Should we use an InstallConfig if given one, otherwise generate?
+   * Add Spec.InstallConfigMap reference.
+   * Likely cannot embed pull secret and ssh key into the InstallConfig in a ConfigMap.
+     * Should we require the InstallConfig in a Secret instead?
+     * Should we template in the secret values on the fly?
```

Comment:
Template them in on the fly.

Reply:
We're actually thinking now that the InstallConfig itself should just be delivered in a Secret rather than a ConfigMap, so no templating should be necessary.